File: catkin_ws/src/hello_pkg/src/sub_hello.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callbackFunc (const std_msgs::String::ConstPtr& msg);

int main(int argc, char **argv)
{
    /*Init node*/
    ros::init(argc, argv, "sub_hello_node");
    /*create a hendler*/
    ros::NodeHandle Handler;

    /*create a subscriber and pass a topic name , message type and callback function
     for handlling the received data*/
    ros::Subscriber sub = Handler.subscribe<std_msgs::String>("hello_topic",1000,callbackFunc);
   
    /*loop*/
    ros::spin();

    return 0;
}

void callbackFunc (const std_msgs::String::ConstPtr& msg)
{
    /*Handle the received message here*/
    ROS_INFO("I heard %s",msg->data.c_str());
}

File: catkin_ws/src/hello_pkg/scripts/sub_hello.py
#! /usr/bin/env python3

from std_msgs.msg import String
import rospy


def main():
    #initialze ros node
    rospy.init_node('sub_hello_node2') 

    #create Subscribe to sub a topic 
    #       topic name   : topic1 
    #       message type : String
    rospy.Subscriber('topic1' , String , callback)
    
    #loop  
    rospy.spin()
    
    
#callback function to handle th ereceived message    
def callback(msg):
    #just to handle the data
    rospy.loginfo("I received %s" % msg.data)  # = print(" --- ")
    
if __name__ == '__main__' :
        main()
